src/Http/OpenApi/test/OpenApiGeneratorTests.cs

```diff
@@ -0,0 +1,814 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+using System.Security.Claims;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Metadata;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Patterns;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Primitives;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.AspNetCore.OpenApi.Tests;
+
+public class OpenApiOperationGeneratorTests
+{
+    [Fact]
+    public void OperationNotCreatedIfNoHttpMethods()
+    {
+        var operation = GetOpenApiOperation(() => { }, "/", Array.Empty<string>());
+
+        Assert.Null(operation);
+    }
+
+    [Fact]
+    public void UsesDeclaringTypeAsOperationTags()
+    {
+        var operation = GetOpenApiOperation(TestAction);
+
+        var declaringTypeName = typeof(OpenApiOperationGeneratorTests).Name;
+        var tag = Assert.Single(operation.Tags);
+
+        Assert.Equal(declaringTypeName, tag.Name);
+
+    }
+
+    [Fact]
+    public void UsesApplicationNameAsOperationTagsIfNoDeclaringType()
+    {
+        var operation = GetOpenApiOperation(() => { });
+
+        var declaringTypeName = nameof(OpenApiOperationGeneratorTests);
+        var tag = Assert.Single(operation.Tags);
+
+        Assert.Equal(declaringTypeName, tag.Name);
+    }
+
+    [Fact]
+    public void AddsRequestFormatFromMetadata()
+    {
+        static void AssertCustomRequestFormat(OpenApiOperation operation)
+        {
+            var request = Assert.Single(operation.Parameters);
+            var content = Assert.Single(request.Content);
+            Assert.Equal("application/custom", content.Key);
+        }
+
+        AssertCustomRequestFormat(GetOpenApiOperation(
+            [Consumes("application/custom")]
+        (InferredJsonClass fromBody) =>
+            { }));
```

Comment:
nit: This seems hard to read with the extra newlines. In this case, there's also an extra closing parenthesis.
```suggestion
        AssertCustomRequestFormat(GetOpenApiOperation(
            [Consumes("application/custom")] (InferredJsonClass fromBody) => { });
```

Reply:
Fixed this up and some other places where inline lambdas with attributes were a bit askew.